### PR TITLE
[0.68] Flatlist: Let ArrowUp/ArrowDown bubble at the top/bottom of list

### PR DESCRIPTION
**Pull request by Saadnajmi**
Copy of #1458 into 0.68-stable

### CODE REVIEW

**Review by lyzhan7 — APPROVED** [review]

File: Libraries/Lists/VirtualizedList.js [hunk 1]
```diff
@@ -1331,9 +1331,21 @@ class VirtualizedList extends React.PureComponent<Props, State> {
     const preferredScrollerStyleDidChangeHandler =
       this.props.onPreferredScrollerStyleDidChange;
 
+    const isFirstRowSelected = this.state.selectedRowIndex === this.state.first;
+    const isLastRowSelected = this.state.selectedRowIndex === this.state.last;
+
+    // Don't pass in ArrowUp/ArrowDown at the top/bottom of the list so that keyboard event can bubble
```
Comment (lyzhan7) [inline Libraries/Lists/VirtualizedList.js:1337]: What does "keyboard event can bubble" mean

File: Libraries/Lists/VirtualizedList.js [hunk 2]
```diff
@@ -1331,9 +1331,21 @@ class VirtualizedList extends React.PureComponent<Props, State> {
...
+    let _validKeysDown = ['Home', 'End'];
```
Comment (amgleitman) [inline Libraries/Lists/VirtualizedList.js:1338]: Why are Home and End always valid but ArrowUp and ArrowDown depend on which row is selected? Not blocking, just curious.

**Review by amgleitman — APPROVED** [review]

File: Libraries/Lists/VirtualizedList.js (hunk 1 above)
Comment (Saadnajmi) [reply Libraries/Lists/VirtualizedList.js:1337]: One day I'll do a brown bag about this, but for now:

https://javascript.info/bubbling-and-capturing

File: Libraries/Lists/VirtualizedList.js (hunk 2 above)
Comment (Saadnajmi) [reply Libraries/Lists/VirtualizedList.js:1338]: Home and End scroll the scrollView, but don't move selection. So regardless of where you are in the list, they have an action they perform. I guess I could remove overriding `Home` if we're at the top of the list too, but I don't know that I have a way to check that like I can check which row is selected.

**Pull request merged**